network.py

```diff
@@ -0,0 +1,91 @@
+from geopandas import GeoDataFrame
+# street nodes, origin nodes, destination_nodes are the different types of nodes
+
+
+class Network:
+    """
+    A road network with nodes `nodes` and edges `edges` weighted with factor `weight_attribute`,
+    projected onto the `projected_crs` coordinate system.
+
+    Internal class, meant to represent a network within the Zonal object.
+    """
+
+    def __init__(self, nodes: GeoDataFrame, edges: GeoDataFrame, projected_crs: str, weight_attribute=None):
+        self.nodes = nodes
+        self.edges = edges
+        self.crs = projected_crs
+        self.weight_attribute = weight_attribute
+        self.G = None  # G represents a networkx graph. To be deprecated
+        return
+
+    def insert_node(self, label: str, layer_name: str, weight: int):
+        """
+        Adds a node with label `label` to the layer `layer_name` with weight `weight`.
+        Maps to insert_nodes_v2 in madina.py
+        """
+        raise NotImplementedError
+
+    def set_node_value(self, idx, label, new_value):
+        """
+        Sets the node at (`idx`, `label`) value in the network to `new_value`.
+        """
+        self.nodes.at[idx, label] = new_value
+        return
+
+    def create_graph(self, light_graph: bool, dense_graph: bool, d_graph: bool):
+        """
+        TODO: Fill out function spec
```

Comment:
@azizH, mind  describing the point of the function in terms of the variables?

Reply:
light_graph: a boolean that enables the creation of a graph that contains ONLY the network nodes and edges (Think: streets and intersections)
dense_graph: a boolean that enables the creation a dense graph: a graphs that contains the network nodes and edges, and also nodes for origins and nodes for destinations, all linked to street nodes and edges by new additional edges (Very expensive graph when you're doing graph-theory calculations: dijkstra, betweenness,...)

d_graph: a boolean that enables the creation of a destination graph (d_graoh), which is to my current implimination of other function, the only used version of the three.. It has all network nodes and edges, and all the destinations, but no origin nodes.. This graph gets updated by the function "update_light_graoh" which is used to add and/or remove needed origins (mostly one origin at a time) to speed up our algorithms (SInce they are mostly a function of one-origin-->many-destinations)

Reply:
Generally, I'd think about keeping this as a network function. The update_light_graph" is in the betweenness_functions.py (because of the timing when I implemented it, not for any other good logical reason)

Reply:
Will do. Thanks for the explanation -- it was very helpful. If `d_graph` is the only one that's used, are the other two boolean parameters necessary?

Reply:
Or to phrase it differently, are those other graphs ever used? If not, we could just make the `create_graph` function a `create_destination_graph` function instead.